src/number.h

```diff
@@ -23,6 +30,15 @@ class Number : public Basic {
     virtual bool is_negative() const = 0;
 
     virtual bool is_positive() const = 0;
+
+    virtual bool is_exact() const { return true; };
+    virtual eval_function get_eval_function() const {
+        throw std::runtime_error("Not Implemented.");
+    };
+    virtual Evaluate* get_eval() const {
+        return nullptr;
+    }
```

Comment:
Either way is fine, whatever is easier to review / get in.

As to pointers, I created #445 with more details.